File: app/src/main/java/com/example/spectrumanalyzer/Texts.java
package com.example.spectrumanalyzer;

/**
 * Created by usr1 on 2017/05/28.
 */

public class Texts {
    private int mMaxNum;
    private int mNum;
    private String[] mText;
    private float[] mX;
    private float[] mY;

    Texts(int maxNum) {
        mMaxNum = maxNum;
        mNum = 0;
        mText = new String[maxNum];
        mX = new float[maxNum];
        mY = new float[maxNum];
    }

    public void init() {
        for (int i = 0; i < mNum; i++) {
            mText[i] = null;
            mX[i] = 0;
            mY[i] = 0;
        }
        mNum = 0;
    }

    public void add(String text, float x, float y) {
        if (mNum < mMaxNum) {
            mText[mNum] = text;
            mX[mNum] = x;
            mY[mNum] = y;
            mNum++;
        }
    }

    public String getText(int index) {
        String ret;
        if ((index >= 0) && (index < mNum)) {
            ret = mText[index];
        } else {
            ret = "";
        }
        return ret;
    }

    public float getX(int index) {
        float ret;
        if ((index >= 0) && (index < mNum)) {
            ret = mX[index];
        } else {
            ret = 0;
        }
        return ret;
    }

    public float getY(int index) {
        float ret;
        if ((index >= 0) && (index < mNum)) {
            ret = mY[index];
        } else {
            ret = 0;
        }
        return ret;
    }

    public int getNum() {
        return mNum;
    }
}

File: app/src/main/java/com/example/spectrumanalyzer/Texts.kt
package com.example.spectrumanalyzer

class Texts internal constructor(private val mMaxNum: Int) {
    private var mNum: Int = 0
    private val mText: Array<String>
    private val mX: FloatArray
    private val mY: FloatArray

    init {
        mText = Array(mMaxNum){""}
        mX = FloatArray(mMaxNum)
        mY = FloatArray(mMaxNum)
    }

    fun init() {
        for (i in 0 until mNum) {
            mText[i] = ""
            mX[i] = 0f
            mY[i] = 0f
        }
        mNum = 0
    }

    fun add(text: String, x: Float, y: Float) {
        if (mNum < mMaxNum) {
            mText[mNum] = text
            mX[mNum] = x
            mY[mNum] = y
            mNum++
        }
    }

    fun getText(index: Int): String {
        val ret: String
        ret = if (index >= 0 && index < mNum) {
            mText[index]
        } else {
            ""
        }
        return ret
    }

    fun getX(index: Int): Float {
        val ret: Float
        ret = if (index >= 0 && index < mNum) {
            mX[index]
        } else {
            0f
        }
        return ret
    }

    fun getY(index: Int): Float {
        val ret: Float
        ret = if (index >= 0 && index < mNum) {
            mY[index]
        } else {
            0f
        }
        return ret
    }

    fun getNum(): Int {
        return mNum
    }
}
